lib/screens/screens_by_alert/behaviour.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Screens.ScreensByAlert.Behaviour do
+  @moduledoc """
+  Behavior for accessing the cache that holds screens by alert.
+
+  - `start_link(opts)` is called to start the backend by the supervisor.
+  - `put_data(screen_id, list(alert_id))` inserts the list of alert_ids visible on a given screen_id
+    to the cache.
+  - `get_screens_by_alert(alert_id)` returns all screen_ids that are currently displaying the given alert_id.
+  - `get_screens_last_updated(screen_id)` returns the timestamp that represents the last time a given screen_id was updated.
+
+  """
+  @type screen_id :: String.t()
+  @type alert_id :: String.t()
+  @type timestamp :: integer()
+  @type timestamped_screen_id :: {screen_id, timestamp}
+
+  @callback start_link(Keyword.t()) :: {:ok, pid()}
+  @callback put_data(screen_id(), list(alert_id())) :: :ok
```

Comment:
Do we need to define a behavior for `put`-ing the `screen_last_updated` data as well, or is that handled elsewhere in the overall solution?

Reply:
Hmm so when the list is empty, we would let the cache expire naturally instead of updating the entry as an empty list ourselves? I think that's what the tech spec indicates so if that's right, then yep I think that makes sense! In that case, would it make sense to make the type spec for `put_data/2` a little more generic?

Reply:
Right, they would either expire naturally or the self-refresh GenServer would refresh/expire them for us. I don't think that we would want to add a screen to `screens_by_alert` unless it was actually showing alerts. 

As for the type spec, the way that it is in the tech spec is unfortunately the best we can do given when the call would occur. We would call `put_data` once per screen refresh, so in that instance we only know the list of alerts for a given screen. We will then have to essentially flip that from `screen_id => list(alert_id)` to `alert_id => list(screen_id)` in the cache module implementation.

Reply:
Hmm yeah I see. Otherwise we'd have to do an additional read _first_ for each alert to get what it currently has as it's list and then append the current screen's id to each of those lists, and then re-`put` them into memcached, which seems pretty inefficient.

So it seems then that we'll be flipping the mapping on read? Sorry, I'm sure this was already discussed at some point but it might've flown over my head or I just missed it.

Reply:
I think the first thing you said is what Jon proposed. In the spec, it says `New cache data from one screen will be distributed across several cache keys, since our updates come in as lists of 0 or more alerts keyed on a screen ID, but the cache is keyed on alert ID), updating TTL metadata at the same time.`. So I think the flip will happen in `put_data` and not on read. I agree that this normally would be super inefficient. But worst case, our screens only show 4 alerts at one time, and that worst case rarely happens. So I don't _think_ it will be too bad.

Reply:
Ohhh hmm gotcha. I suppose even if it's a little inefficient, it's still better than flipping the map on read now that I think about it. If we did that, I think we'd have to iterate through the entire cache while checking for the alert_id(s) on every request.

I guess my main concern then would just be readability since it might be a little obfuscating to have the function definition read as "put this list of alert ids for this screen id" when in reality, it's doing the reverse. But that's not a big deal I think as long as it's well documented

Reply:
That's a good point. I'll add some comments to the behaviour.